### ISSUE TITLE
question about training moco

### ISSUE THREAD

**Issue by UcanSee** [open-mmlab/mmselfsup#28]
I used the configs/selfsup/moco/r50_v2.py to train moco, but I found the loss increase to 9.60 at start and then decrease slowly, is it normal?

**Comment by XiaohangZhan** [comment]
Normal. In my case, it increased to 10.3 and then decreased slowly. It reached 6.57 at last.

**Comment by UcanSee** [comment]
I‘ve got it， thank you！

**Comment by UcanSee** [comment]
I‘ve got it， thank you！

**Comment by a1600012888** [comment]
Hi, has there anyone studied the influence of batch size for mocov2?  I only did one simple experiment without any tricks. 
I increased the batch size to 2048(keep the batch size in single gpu as 64) and linearly increase the initial learning-rate as well. 
But the final training loss only reached 6.594, and the accuracy on ImageNet under linear protocol only reached 62.9%.

**Comment by XiaohangZhan** [comment]
I did not try 2048. The moco-v2 paper has no such experiment either. With a large batch size, I suppose LARS rather than SGD is better.

**Comment by UcanSee** [comment]
I downloaded the pretrained model moco_r50_v2-58f10cfe.pth from model zoo, and run semi-supervied experiment by 'r50_lr0_01_head100.py' on ImageNet  with 10% training data, but I just got 59.47% top-1 accuracy on val set, not as well as 61.64% reported in the repo, is there something wrong when I conduct the experiment?

**Comment by XiaohangZhan** [comment]
Please show me your command or script.

**Comment by UcanSee** [comment]
the command is ''bash tools/dist_train.sh configs/benchmarks/semi_classification/imagenet_10percent/r50_lr0_01_head100.py 4 --work_dir work_dirs/r50_semi_supervison_10_percent_mocov2/ --seed 0'',  and I just changed the value of 'pretrained' in config from None to path of moco_r50_v2-58f10cfe.pth.

**Comment by XiaohangZhan** [comment]
> the command is ''bash tools/dist_train.sh configs/benchmarks/semi_classification/imagenet_10percent/r50_lr0_01_head100.py 4 --work_dir work_dirs/r50_semi_supervison_10_percent_mocov2/ --seed 0'',  and I just changed the value of 'pretrained' in config from None to path of moco_r50_v2-58f10cfe.pth.

The command is wrong, please follow docs/GETTING_STARTED.md: ImageNet Semi-Supervised Classificatio.

**Comment by UcanSee** [comment]
Is there any different between this two command？I think no difference if I choose correct config file and pretrained pth file.

**Comment by XiaohangZhan** [comment]
No, there is no difference. Could please first try to follow the standard way to yield a result, so that we can localize the cause?

**Comment by a1600012888** [comment]
Hi, Xiaohang.  I checked the mocov2 official code. 
I find some minor difference in the data preprocessing stages.
The hyperparam for ColorJittering in https://github.com/facebookresearch/moco/blob/78b69cafae80bc74cd1a89ac3fb365dc20d157d3/main_moco.py#L228 is (0.4, 0.4, 0.4, 0.1)
While in your repo https://github.com/open-mmlab/OpenSelfSup/blob/master/configs/selfsup/moco/r50_v2.py#L41, it is (0.4, 0.4, 0.4, 0.4)

**Comment by XiaohangZhan** [comment]
> Hi, Xiaohang. I checked the mocov2 official code.
> I find some minor difference in the data preprocessing stages.
> The hyperparam for ColorJittering in https://github.com/facebookresearch/moco/blob/78b69cafae80bc74cd1a89ac3fb365dc20d157d3/main_moco.py#L228 is (0.4, 0.4, 0.4, 0.1)
> While in your repo https://github.com/open-mmlab/OpenSelfSup/blob/master/configs/selfsup/moco/r50_v2.py#L41, it is (0.4, 0.4, 0.4, 0.4)

Thank you so much to find out that. We will re-run the experiment immediately.

**Comment by UcanSee** [comment]
hello， I want to know is there any improvement if do not use SyncBN (only use  BN)  in backbone and neck when training an object detection model?

**Comment by XiaohangZhan** [comment]
We just follow moco's setting for object detection. Changing evaluation setting is not applicable since it will result in unfair comparisons. We also never try to change SyncBN to BN in detection.

**Comment by mathfinder** [comment]
> > Hi, Xiaohang. I checked the mocov2 official code.
> > I find some minor difference in the data preprocessing stages.
> > The hyperparam for ColorJittering in https://github.com/facebookresearch/moco/blob/78b69cafae80bc74cd1a89ac3fb365dc20d157d3/main_moco.py#L228 is (0.4, 0.4, 0.4, 0.1)
> > While in your repo https://github.com/open-mmlab/OpenSelfSup/blob/master/configs/selfsup/moco/r50_v2.py#L41, it is (0.4, 0.4, 0.4, 0.4)
> 
> Thank you so much to find out that. We will re-run the experiment immediately.

How is the progress of the rerun experiment? Does this difference have a significant impact on the results of the experiment?

**Comment by XiaohangZhan** [comment]
The result has not come out yet. We need about 1 more week.

**Comment by a1600012888** [comment]
> Hi, has there anyone studied the influence of batch size for mocov2? I only did one simple experiment without any tricks.
> I increased the batch size to 2048(keep the batch size in single gpu as 64) and linearly increase the initial learning-rate as well.
> But the final training loss only reached 6.594, and the accuracy on ImageNet under linear protocol only reached 62.9%.

The results on  ImageNet under linear protocol was wrong, I fixed some bugs on this group of experiments, and it reached 66.2%.  For more results on large batch and mixed precision training, please see https://github.com/open-mmlab/OpenSelfSup/issues/35

**Comment by XiaohangZhan** [comment]
> > > Hi, Xiaohang. I checked the mocov2 official code.
> > > I find some minor difference in the data preprocessing stages.
> > > The hyperparam for ColorJittering in https://github.com/facebookresearch/moco/blob/78b69cafae80bc74cd1a89ac3fb365dc20d157d3/main_moco.py#L228 is (0.4, 0.4, 0.4, 0.1)
> > > While in your repo https://github.com/open-mmlab/OpenSelfSup/blob/master/configs/selfsup/moco/r50_v2.py#L41, it is (0.4, 0.4, 0.4, 0.4)
> > 
> > 
> > Thank you so much to find out that. We will re-run the experiment immediately.
> 
> How is the progress of the rerun experiment? Does this difference have a significant impact on the results of the experiment?

It has reached 67.6 up to now even though the linear evaluation has not finished. It's proven that hue=0.1 really makes the difference!

**Comment by mathfinder** [comment]
> > > > Hi, Xiaohang. I checked the mocov2 official code.
> > > > I find some minor difference in the data preprocessing stages.
> > > > The hyperparam for ColorJittering in https://github.com/facebookresearch/moco/blob/78b69cafae80bc74cd1a89ac3fb365dc20d157d3/main_moco.py#L228 is (0.4, 0.4, 0.4, 0.1)
> > > > While in your repo https://github.com/open-mmlab/OpenSelfSup/blob/master/configs/selfsup/moco/r50_v2.py#L41, it is (0.4, 0.4, 0.4, 0.4)
> > > 
> > > 
> > > Thank you so much to find out that. We will re-run the experiment immediately.
> > 
> > 
> > How is the progress of the rerun experiment? Does this difference have a significant impact on the results of the experiment?
> 
> It has reached 67.6 up to now even though the linear evaluation has not finished. It's proven that hue=0.1 really makes the difference!

Which neck did you use to reach 67.69? Non-linear neck v1 or BYOL's neck?

**Comment by XiaohangZhan** [comment]
NonLinearNeckV1, as in the config: r50_v2.py

**Comment by mathfinder** [comment]
> > > > Hi, Xiaohang. I checked the mocov2 official code.
> > > > I find some minor difference in the data preprocessing stages.
> > > > The hyperparam for ColorJittering in https://github.com/facebookresearch/moco/blob/78b69cafae80bc74cd1a89ac3fb365dc20d157d3/main_moco.py#L228 is (0.4, 0.4, 0.4, 0.1)
> > > > While in your repo https://github.com/open-mmlab/OpenSelfSup/blob/master/configs/selfsup/moco/r50_v2.py#L41, it is (0.4, 0.4, 0.4, 0.4)
> > > 
> > > 
> > > Thank you so much to find out that. We will re-run the experiment immediately.
> > 
> > 
> > How is the progress of the rerun experiment? Does this difference have a significant impact on the results of the experiment?
> 
> It has reached 67.6 up to now even though the linear evaluation has not finished. It's proven that hue=0.1 really makes the difference!

The result is that the gaussian_blur's kernel size (size=23) is removed, right?

**Comment by XiaohangZhan** [comment]
No, 67.6 comes from cv2 gaussian blur that requires a kernel size (same as simclr). We found that it is much slower than PIL and moco v2 actually uses PIL. Hence we switch it to PIL. Since 67.6 reproduced the official result, we believe it won't affect a lot whether using cv2 or PIL.

**Comment by NickChang97** [comment]
> Hi, has there anyone studied the influence of batch size for mocov2? I only did one simple experiment without any tricks.
> I increased the batch size to 2048(keep the batch size in single gpu as 64) and linearly increase the initial learning-rate as well.
> But the final training loss only reached 6.594, and the accuracy on ImageNet under linear protocol only reached 62.9%.

Hi, did you try to use LARS when batch size set 2048 . If it is, can you share your result about that. Thanks in advance
